components/arm/client.go

```diff
@@ -112,18 +117,45 @@ func (c *client) Stop(ctx context.Context, extra map[string]interface{}) error {
 }
 
 func (c *client) ModelFrame() referenceframe.Model {
-	// TODO(erh): this feels wrong
-	return nil
+	return c.model
 }
 
 func (c *client) CurrentInputs(ctx context.Context) ([]referenceframe.Input, error) {
-	return nil, errArmClientInputsNotSupport
+	resp, err := c.JointPositions(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+	if c.model == nil {
+		return nil, errArmClientModelNotValid
+	}
+	return c.model.InputFromProtobuf(resp), nil
 }
 
 func (c *client) GoToInputs(ctx context.Context, goal []referenceframe.Input) error {
-	return errArmClientInputsNotSupport
+	if c.model == nil {
+		return errArmClientModelNotValid
+	}
+	return c.MoveToJointPositions(ctx, c.model.ProtobufFromInput(goal), nil)
 }
 
 func (c *client) DoCommand(ctx context.Context, cmd map[string]interface{}) (map[string]interface{}, error) {
 	return generic.DoFromConnection(ctx, c.conn, c.name, cmd)
 }
+
+func getModel(ctx context.Context, r robotpb.RobotServiceClient, name string) referenceframe.Model {
+	resp, err := r.FrameSystemConfig(ctx, &robotpb.FrameSystemConfigRequest{})
+	if err != nil {
+		return nil
+	}
+	cfgs := resp.GetFrameSystemConfigs()
+	for _, cfg := range cfgs {
+		part, err := config.ProtobufToFrameSystemPart(cfg)
```

Comment:
Can we check names of things without doing a conversion here?

Reply:
Besides getting the name this function lets us parse the model anyway so I think this is necessary

Reply:
There's no need to call the protobuf convert on every single `cfg`.

`Name` is literally a field on the cfg struct: https://github.com/viamrobotics/api/blob/7306c122f88b65c53a576b030e039523cdc0628b/robot/v1/robot.pb.go#L28

So you can check `if cfg.Name == name` and only call ProtobufToFrameSystemPart to get the model in that case.

Reply:
Ah sorry I didn't follow the first time.  I will change this